[MM-35890] Replaces log.Print with g.LogError

willypuzzle opened the pull request:
#### Summary
Replaces instances of `log.Print` with `g.LogError` in the plugin client RPC to provide more structured logging and include error context. This change enhances error reporting and debugging capabilities.

#### Ticket Link
Fixes https://github.com/mattermost/mattermost/issues/17650
Jira https://mattermost.atlassian.net/browse/MM-35890


#### Release Note
```release-note
NONE
```

<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

* **Refactor**
  * Centralized and standardized error logging across the plugin RPC and DB layers, replacing ad-hoc prints with structured logging for clearer, consistent diagnostics.
  * Logging now routes through the shared plugin API to unify audit, debug, info, warn and error messages.
  * No public API signatures were changed.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

willypuzzle committed:
Replaces log.Print with g.LogError

Replaces instances of `log.Print` with `g.LogError` in the plugin
client RPC to provide more structured logging and include error
context. This change enhances error reporting and debugging
capabilities.

Related to: MM-35890

mattermost-build commented:
Hello @willypuzzle,

Thanks for your pull request! A Core Committer will review your pull request soon. For code contributions, you can learn more about the review process [here](https://developers.mattermost.com/contribute/getting-started/code-review/).

hanzei reviewed (changes requested):
Thanks for the PR 👍 

Two comments:
1. I think you missed a couple of cases in `client_rpc.go`. e.g. L920 doesn't get fixed
2. `client_rpc_generated.go` also has a calls to `log.Printf`. Would you be open to migrating them as well? That will require updating the generation code in `public/plugin/interface_generator/main.go`.

willypuzzle committed:
Improves plugin RPC client logging

Replaces direct logging with `g.LogError` in the plugin RPC client to include error context using `mlog.Err(err)`.
This provides more structured and informative error messages for debugging purposes.

Relates to MM-35890

willypuzzle commented:
> Thanks for the PR 👍
> 
> Two comments:
> 
> 1. I think you missed a couple of cases in `client_rpc.go`. e.g. L920 doesn't get fixed
> 2. `client_rpc_generated.go` also has a calls to `log.Printf`. Would you be open to migrating them as well? That will require updating the generation code in `public/plugin/interface_generator/main.go`.

@hanzei 

how can I update the template? in the file I read `"// Code generated by "make pluginapi""` but I call `make pluginapi` and the file doesn't change

willypuzzle committed:
Updates RPC call error logging

Updates the RPC client to use structured logging for RPC call errors.

This improves the context provided by the logs, making debugging easier.

Relates to MM-35890

willypuzzle committed:
Merge branch 'master' into MM-35890-Revisit-plugin-RPC-client-log-levels

willypuzzle committed:
Replaces plugin RPC client logs with LogError

Replaces `log.Printf` with `g.LogError` to ensure proper logging of errors originating from RPC calls within the plugin client.
This change ensures that errors include context and are handled according to the configured logging level.

hanzei reviewed (commented):
That is much better, thanks!

It would be nice to also get the code in `public/plugin/db_rpc.go` migrated as part of this PR. Would you be interested in also doing that?

mattermost-build commented:
This PR has been automatically labelled "stale" because it hasn't had recent activity.
A core team member will check in on the status of the PR to help with questions.
Thank you for your contribution!

willypuzzle committed:
Merge branch 'master' into MM-35890-Revisit-plugin-RPC-client-log-levels

willypuzzle committed:
Removes unused log import

Removes an unnecessary log import from the interface generator, cleaning up dependencies.

Relates to MM-35890

willypuzzle commented:
> That is much better, thanks!
> 
> It would be nice to also get the code in `public/plugin/db_rpc.go` migrated as part of this PR. Would you be interested in also doing that?

@hanzei How can I proceed? I don't see andy db.log* in the struct. Can I call mlog directly?

willypuzzle committed:
Merge branch 'master' into MM-35890-Revisit-plugin-RPC-client-log-levels

hanzei commented:
@willypuzzle Can we give `dbRPCClient` a point to `dbRPCClient` so I has access to `apiRPCClient`? Or even better, add a small logging interface to `dbRPCClient`?

willypuzzle committed:
Adds RPC client logging methods

Adds `LogError`, `LogWarn`, `LogInfo`, and `LogDebug` methods to the RPC client.

These methods facilitate structured logging from the plugin to the server
through the RPC interface.

Relates to MM-35890

willypuzzle committed:
Replaces log.Printf with db.LogError

This change replaces instances of `log.Printf` with `db.LogError` calls within the db_rpc.go file.

This ensures consistent and structured logging using the server's logging framework, including improved context and error handling.

willypuzzle committed:
Merge branch 'master' into MM-35890-Revisit-plugin-RPC-client-log-levels

willypuzzle committed:
Simplifies dbRPCClient log level methods.

Updates the dbRPCClient's LogError, LogWarn, LogInfo and LogDebug methods to use the receiver `db` instead of `g`, to improve readability and consistency.

MM-35890

willypuzzle committed:
Merge branch 'master' into MM-35890-Revisit-plugin-RPC-client-log-levels

willypuzzle committed:
Simplifies plugin logging via direct API calls

Refactors the dbRPCClient to use the API interface directly
for logging, removing the need for custom RPC calls and
associated data structures for each log level.

This streamlines the logging process and reduces code
duplication, improving maintainability.

Related to MM-35890

hanzei reviewed (approved):
LGTM 👍

hanzei commented:
@DHaussermann In order to test the changes, we need to compile a new version of a plugin using the newly introduced code from this PR. Are you familiar with that process? Happy to discuss the concrete testing steps during our 1-1.

mattermost-build committed:
Merge branch 'master' into MM-35890-Revisit-plugin-RPC-client-log-levels

mattermost-build committed:
Merge branch 'master' into MM-35890-Revisit-plugin-RPC-client-log-levels

willypuzzle committed:
Updates RPC client error logging

Refines how errors are passed to the `LogError` function by providing `mlog.Err(err)` as a separate argument rather than formatting it into a string.

This change allows the logger to better process and enrich structured error information from `mlog.Err` objects, improving overall error reporting consistency and detail.

Relates to MM-35890

willypuzzle committed:
Standardizes plugin RPC client error logging

Refactors RPC client error logging to use the structured `g.LogError` method, enhancing consistency and enabling better log analysis with `mlog.Err`. This aligns with efforts to improve plugin RPC client log levels.

Relates to MM-35890

willypuzzle commented on server/public/plugin/client_rpc.go:
```diff
@@ -1180,7 +1181,7 @@ func (g *apiRPCClient) InstallPlugin(file io.Reader, replace bool) (*model.Manif
+		g.LogError("RPC call InstallPlugin to plugin failed.", mlog.Err(err))
 	}
 
 	return _returns.A, _returns.B
```
@hanzei I think coderabbit got wrong, because I think that `_returns.A`, `_returns.B` are properly set when returned. Even in error situation.

willypuzzle commented:
/update-branch

mattermost-build committed:
Merge branch 'master' into MM-35890-Revisit-plugin-RPC-client-log-levels
